web/modules/custom/dbcdk_community/src/Form/ProfileEditForm.php

```diff
@@ -24,79 +24,50 @@
 class ProfileEditForm extends FormBase implements ContainerInjectionInterface {
 
   /**
-   * The DBCDK Community Service Profile API.
+   * The profile repository to use.
    *
-   * @var \DBCDK\CommunityServices\Api\ProfileApi $profileApi
+   * @var \Drupal\dbcdk_community\Profile\ProfileRepository $profileRepository
    */
-  protected $profileApi;
+  protected $profileRepository;
 
   /**
    * The Community Service Profile object from the username context.
    *
-   * @var \DBCDK\CommunityServices\Model\Profile $profile
+   * @var Profile $profile
    */
   protected $profile;
 
   /**
-   * Array of all defined community roles.
-   *
-   * @var \DBCDK\CommunityServices\Model\CommunityRole[] $communityRoles
-   */
-  protected $communityRoles;
-
-  /**
    * Creates a Profile Edit Form instance.
    *
-   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
-   *   The DBCDK Community Service Profile API.
-   * @param \DBCDK\CommunityServices\Model\Profile $profile
+   * @param \Drupal\dbcdk_community\Profile\ProfileRepository $profile_repository
+   *   The profile repository to use.
+   * @param \DBCDK\CommunityServices\Model\Profile|\Drupal\dbcdk_community\Profile\Profile $profile
```

Comment:
You are totally right. It should always be a subclassed/enriched Profile passed as an argument.